<a href="https://colab.research.google.com/github/chandani1001/my-DataSciencePractice/blob/main/Chi_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Chi square test

Application in 2 types of situations.

*   Goodness   
*   Independence

Chi Square test:

Formula
X^2= ∑ (Oi-Ei)^2/Ei

X^2	=	chi squared
Oi	=	observed value
Ei	=	expected value



**Example 1:**

A fair die is rolled 120 times and the following results are obtained:
    
Face 1: 22 times

Face 2: 17 times

Face 3: 20 times

Face 4: 26 times

Face 5: 22 times

Face 6: 13 times

Test at a 5% level of significance whether the die is fair.

soln:
expected times be 20 per every face

H0 dide is fair,
Ha die is not fair
let the alpha = 0.05 as 5% significance & it is 1 tailed normal distribution.

Also ask for df(degree of freedom)=n-1=6-1=5

now from the chi square table:

chi_table=11.07

now chi_cal=∑ (Oi-Ei)^2/Ei

In [9]:
import numpy as np
import scipy.stats as st

In [10]:
ob=np.array([22,17,20,26,22,13])
ex=np.array([20,20,20,20,20,20])

In [12]:
chi_table=st.chi2.ppf(0.95,5)
chi_table

np.float64(11.070497693516351)

In [11]:
chi_cal=np.sum(np.square(ob-ex)/ex)
chi_cal

np.float64(5.1000000000000005)

Chi_cal= 5.100

Decision Rule:

If χ²_cal > χ²_table → Reject H₀

If χ²_cal ≤ χ²_table → Fail to Reject H₀




In [13]:
if chi_cal > chi_table:
    print("Reject H₀ → The die is not fair.")
else:
    print("Fail to Reject H₀ → The die is fair.")

Fail to Reject H₀ → The die is fair.


**Example 2:**

A study was conducted to investigate whether there is a relationship between gender and the preferred genre of music. A sample of 235 people was selected, and the data collected is shown below. Test at a 5% level of significance whether there is a significant association between gender and music preference.

sn.   Pop,Hip Hop,Classical,Rock

Male:40,  45,   25,      10

Female:35, 30,  20,       30



Soln:
H0: No association
Ha: association

alpha=0.05
df=(row-1)(col-1)=(2-1)(4-1)=3

In [14]:
import scipy.stats as st
import numpy as np

In [15]:
chi_table= st.chi2.ppf(1-0.05,3)
chi_table

np.float64(7.814727903251179)

In [16]:
row1=np.array([40, 45, 25, 10])
row2=np.array([35, 30, 20, 30])

In [22]:
sum_r1=np.sum(row1)
sum_r2=np.sum(row2)
sum_r1,sum_r2
sum_row=np.array([sum_r1,sum_r2])
sum_row

array([120, 115])

In [19]:
sum_col= row1+ row2
sum_col

array([75, 75, 45, 40])

now we are finding the expected value.

In [26]:
exp=[]
for i in sum_row:
  for j in sum_col:
    value=(i*j)/235
    exp.append(value)

In [27]:
exp

[np.float64(38.297872340425535),
 np.float64(38.297872340425535),
 np.float64(22.97872340425532),
 np.float64(20.425531914893618),
 np.float64(36.702127659574465),
 np.float64(36.702127659574465),
 np.float64(22.02127659574468),
 np.float64(19.574468085106382)]

In [28]:
obj= np.array([40, 45, 25, 10,35, 30, 20, 30])
obj

array([40, 45, 25, 10, 35, 30, 20, 30])

In [33]:
chi_cal=np.sum(np.square(obj-exp)/exp)
chi_cal

np.float64(13.788747987117553)

In [34]:
if chi_cal > chi_table:
    print("Reject H₀ → There is a significant association between gender and music preference.")
else:
    print("Fail to Reject H₀ → There is no significant association between gender and music preference.")

Reject H₀ → There is a significant association between gender and music preference.
